levels/level_manager_test.go

```diff
@@ -2082,3 +2082,173 @@ func TestRegisterDeadVersionRangeDedup(t *testing.T) {
 
 	require.Equal(t, OverlappingTables(expected), tables)
 }
+func TestContainsTable(t *testing.T) {
+	populatedSeg := &segment{
+		format: 26,
+		tableEntries: []*TableEntry{
+			{
+				SSTableID:  []byte("sstableid1"),
+				RangeStart: []byte("rangestart1"),
+				RangeEnd:   []byte("rangeend1"),
+			},
+			{
+				SSTableID:  []byte("sstableid2"),
+				RangeStart: []byte("rangestart2"),
+				RangeEnd:   []byte("rangeend2"),
+			},
+		},
+	}
+	emptySeg := &segment{
+		format:       26,
+		tableEntries: []*TableEntry{},
+	}
+	var testCases = []struct {
+		name     string
+		seg      *segment
+		tabID    []byte
+		expected bool
+	}{
+		{
+			name:     "sstableid1",
+			seg:      populatedSeg,
+			tabID:    []byte("sstableid1"),
+			expected: true,
+		},
+		{
+			name:     "sstableid2",
+			seg:      populatedSeg,
+			tabID:    []byte("sstableid2"),
+			expected: true,
+		},
+		{
+			name:     "non-existing table id",
+			seg:      populatedSeg,
+			tabID:    []byte("sstableid3"),
+			expected: false,
+		},
+		{
+			name:     "empty table id",
+			seg:      emptySeg,
+			tabID:    []byte(""),
+			expected: false,
+		},
+		{
+			name:     "table id less than first entry",
+			seg:      populatedSeg,
+			tabID:    []byte("aaaaa"),
+			expected: false,
+		},
+		{
+			name:     "table id greater than last entry",
+			seg:      populatedSeg,
+			tabID:    []byte("zzzzz"),
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := containsTable(tc.seg, tc.tabID)
+			if result != tc.expected {
+				t.Errorf("expected %v, got %v", tc.expected, result)
+			}
+		})
+	}
+}
+
+func TestGetSegmentForRegistration(t *testing.T) {
+	segmentEntries := []segmentEntry{
+		{rangeStart: []byte("aaa"), rangeEnd: []byte("ccc")},
+		{rangeStart: []byte("ddd"), rangeEnd: []byte("fff")},
+		{rangeStart: []byte("ggg"), rangeEnd: []byte("iii")},
+	}
+
+	// Test cases
+	testCases := []struct {
+		name         string
+		registration RegistrationEntry
+		expected     int
+	}{
+		{name: "no valid insertion point", registration: RegistrationEntry{KeyStart: []byte("b"), KeyEnd: []byte("e")}, expected: -1},
```

Comment:
I think it would be good here to test all the boundary conditions - e.g. when key start / key end are exactly on the range start / end of existing entries, and when they're just one away.

Reply:
I assumed by "one away" you meant in the sense of `[]byte("aaa")` vs `[]byte("aab")`